### PR TITLE
feat: [PL-24488]: Add emailId as part of the ApproveOrReject GraphQL request payload

### PR DESCRIPTION
**Pull request by rafaelmart1ns**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/37833)

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 3 Issues 
* Code issue 3 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/7752dbb3099d2dba3b5d37f5931a5e4c/pull-request/37833)

**Review by Fernando-Dourado — CHANGES_REQUESTED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37833)* status: 6 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[`380-cg-graphql/src/main/java/software/wings/graphql/datafetcher/approval/ApproveOrRejectApprovalsDataFetcher.java` line 90 at r2](https://reviewable.io/reviews/harness/harness-core/37833#-NCf0HnL4TVsF4UC7pKq:-NCf0HnL4TVsF4UC7pKr:bq8vh1z) ([raw file](https://github.com/harness/harness-core/blob/5719d0511299647ffaf9852b73b49aaf1d19b95d/380-cg-graphql/src/main/java/software/wings/graphql/datafetcher/approval/ApproveOrRejectApprovalsDataFetcher.java#L90)):*
> ```Java
>     ApiKeyEntry apiKeyEntry = graphQLContext.get("apiKeyEntry");
> 
>     if (approveOrRejectApprovalsInput.getUserEmail() != null
> ```

Please, evaluate the condition one time and save it to a boolean to check in the second if condition.

**Review by rafaelmart1ns — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37833)* status: 6 file(s) to review. 1 change requests. 1. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`380-cg-graphql/src/main/java/software/wings/graphql/datafetcher/approval/ApproveOrRejectApprovalsDataFetcher.java` line 90 at r2](https://reviewable.io/reviews/harness/harness-core/37833#-NCf0HnL4TVsF4UC7pKq:-NCf7RSTDUpVUSCOLxZ8:b-bmgb9m) ([raw file](https://github.com/harness/harness-core/blob/5719d0511299647ffaf9852b73b49aaf1d19b95d/380-cg-graphql/src/main/java/software/wings/graphql/datafetcher/approval/ApproveOrRejectApprovalsDataFetcher.java#L90)):*
<details><summary><i>Previously, Fernando-Dourado (Fernando Dourado) wrote…</i></summary><blockquote>

Please, evaluate the condition one time and save it to a boolean to check in the second if condition.
</blockquote></details>

it fixed.

**Review by Fernando-Dourado — APPROVED** [review]
> Reviewed 3 of 6 files at r1, 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37833)* status: 6 file(s) to review. 1 of 2 approvals obtained.

**Review by rafaelmart1ns — COMMENTED** [review]
> Reviewed 4 of 6 files at r1, 1 of 1 files at r2, 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37833)* status: 6 file(s) to review. 1 of 2 approvals obtained.

**Review by Fernando-Dourado — APPROVED** [review]
> Reviewed 1 of 6 files at r1, 1 of 1 files at r2.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37833)* status: 6 file(s) to review. 1 of 2 approvals obtained.

**Review by Fernando-Dourado — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37833)* status: 6 file(s) to review. 1 of 2 approvals obtained.

File: 380-cg-graphql/src/main/java/software/wings/graphql/datafetcher/approval/ApproveOrRejectApprovalsDataFetcher.java [hunk 1]
```diff
@@ -72,15 +84,31 @@ protected QLApproveOrRejectApprovalsPayload mutateAndFetch(
...
 
+    boolean hasNotEmailAndFFEnabled = (approveOrRejectApprovalsInput.getUserEmail() != null)
+        && featureFlagService.isEnabled(
+            FeatureName.SPG_ENABLE_EMAIL_TO_APPROVE_OR_REJECT_GRAPHQL, execution.getAccountId());
     GraphQLContext graphQLContext = mutationContext.getDataFetchingEnvironment().getContext();
     ApiKeyEntry apiKeyEntry = graphQLContext.get("apiKeyEntry");
 
+    if (hasNotEmailAndFFEnabled) {
+      if (!isUserPresentInApiKey(approveOrRejectApprovalsInput.getUserEmail(), apiKeyEntry)) {
+        throw new InvalidRequestException("User email does not valid or your user cannot access", WingsException.USER);
```
Comment (vcalasansh) [inline 380-cg-graphql/src/main/java/software/wings/graphql/datafetcher/approval/ApproveOrRejectApprovalsDataFetcher.java:95]: Up to you, but I would suggest replacing "does not valid" for "is invalid" .

File: 380-cg-graphql/src/main/java/software/wings/graphql/datafetcher/approval/ApproveOrRejectApprovalsDataFetcher.java [hunk 2]
```diff
@@ -72,15 +84,31 @@ protected QLApproveOrRejectApprovalsPayload mutateAndFetch(
       throw new InvalidRequestException("Execution does not exist or access is denied", WingsException.USER);
     }
 
+    boolean hasNotEmailAndFFEnabled = (approveOrRejectApprovalsInput.getUserEmail() != null)
```
[the remark this answers is not in the document]
Comment (rafaelmart1ns) [reply 380-cg-graphql/src/main/java/software/wings/graphql/datafetcher/approval/ApproveOrRejectApprovalsDataFetcher.java]: Yes it should. done.

File: 380-cg-graphql/src/main/java/software/wings/graphql/datafetcher/approval/ApproveOrRejectApprovalsDataFetcher.java (hunk 1 above)
Comment (rafaelmart1ns) [reply 380-cg-graphql/src/main/java/software/wings/graphql/datafetcher/approval/ApproveOrRejectApprovalsDataFetcher.java]: Suggestion accepted. done.

**Review by vcalasansh — APPROVED** [review]
> Reviewed 2 of 6 files at r1, 1 of 1 files at r2, 3 of 3 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37833)* status: 3 file(s) to review. 1 of 2 approvals obtained.

**Review by Fernando-Dourado — APPROVED** [review]
> Reviewed 3 of 3 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37833)* status: :shipit: complete! ✓

**Pull request merged**